### PR TITLE
Remove user panel mutation + fix panel UI overflow

### LINKED ISSUE #415: Profile side panel remove user mutation call
Add apollo client call in the user profile side panel in the user management page. The mutation handler should be attached to the remove user button or confirm modal as appropriate.

## Mockup

Figma: https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=1643%3A48828

## Technical:

- Create a mutation call in `/admin/users` page to remove a user (volunteer or employee) by id
- Use the `deleteVolunteerUserByEmail` and `deleteEmployeeUserByEmail` mutations
- Please create any necessary graphql request types if they do not exists

### LINKED ISSUE #574: User management side panel buttons not displaying properly
**Role:** Super Admin

**Route:** https://sistering-dev.web.app/admin/users

**Description:** When volunteers have branch tags attached to them, the buttons on the sidebar do not overflow correctly. Scrolling is also disabled on the sidebar, so the buttons cannot be read.

**Steps to Reproduce:** Assign a volunteer to some branches and view their profile

**Expected:** Buttons should be visible

**Actual:**

### PR DESCRIPTION
**Pull request by lambo-liu**
## Ticket link

Closes #415 
Closes #574 

## Implementation description
* Modified prisma schema to cascade delete a user's signups.
* Hooked up the "Remove user" button with the remove volunteer/admin mutation depending on the user's role
* IMPORTANT: Do NOT delete a seed user! The remove user mutation also removes them from Firebase.
* Refetched table when a user is removed.
* Fixed UI overflow on side panel when many tags are present.

## Steps to test
1. Run `npx prisma migrate dev` for the latest table changes
2. Create a new admin or volunteer account (Do not use seed accounts!!)
3. Remove the user using a super admin account
4. Check that the user is removed from SQL tables and Firebase

## What should reviewers focus on?
* Correct prisma schema updates

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/admin/user/AdminUserManagementPage.tsx [hunk 1]
```diff
@@ -223,7 +223,7 @@ const AdminUserManagementPage = (): React.ReactElement => {
     AdminUserManagementTableTab.Volunteers,
   );
 
-  const { loading, error } = useQuery(USERS, {
+  const { loading, error, refetch } = useQuery(USERS, {
```
Comment (MatoPlus) [inline frontend/src/components/pages/admin/user/AdminUserManagementPage.tsx:226]: Let's avoid drilling and use the `refetchQueries` option for mutations for auto-refetch.

File: frontend/src/components/admin/users/ProfileDrawer.tsx [hunk 2]
```diff
@@ -61,80 +75,97 @@ const ProfileDrawer = ({
...
   selectedBranches,
   handleBranchMenuItemClicked,
+  refetchUsers,
 }: ProfileDrawerProps): React.ReactElement => {
   const [openDeleteModal, setOpenDeleteModal] = useState(false);
+  const [deleteUserByEmail] = useMutation(
+    isVolunteer
+      ? DELETE_VOLUNTEER_USER_BY_EMAIL
+      : DELETE_EMPLOYEE_USER_BY_EMAIL,
+  );
```
Comment (MatoPlus) [inline frontend/src/components/admin/users/ProfileDrawer.tsx:85]:
```suggestion
  const [deleteUserByEmail] = useMutation(
    isVolunteer
      ? DELETE_VOLUNTEER_USER_BY_EMAIL
      : DELETE_EMPLOYEE_USER_BY_EMAIL,
          {
      refetchQueries: [
        "AdminUserManagementPage_Users",
      ],
    },
  );
```

File: frontend/src/components/pages/admin/user/AdminUserManagementPage.tsx (hunk 1 above)
Comment (MatoPlus) [reply frontend/src/components/pages/admin/user/AdminUserManagementPage.tsx:226]: I've tested the current implementation with drilling and it doesn't seem like the user data is updated until page refresh

**Review by MatoPlus — CHANGES_REQUESTED** [review]
Made some suggestions. Tested, works as intended. Should be good to merge

File: frontend/src/components/admin/users/ProfileDrawer.tsx (hunk 2 above)
Comment (lambo-liu) [reply frontend/src/components/admin/users/ProfileDrawer.tsx]: Good call! Made the changes, feel free to take another look.

**Review by MatoPlus — APPROVED** [review]
LTGM, thanks for getting this done!

**Pull request merged**